### PR TITLE
Improved recording capabilities and configurability.

### PR DESCRIPTION
**Pull request by shadielhajj**
Added several configuration options to make the recording subsystem more flexible and enable high-quality recordings.
Also moved the stat widget to the bottom left, so it doesn't overlap with the GUI.
The new options are documented in the README under Recording Capabilities.
All new options have defaults which match the current settings.

* Added `shader-toy.forceResolution` (force canvas resolution),
* Added `shader-toy.recordVideoContainer` (set video file container),
* Added `shader-toy.recordVideoCodec` (set video codec),
* Added `shader-toy.recordVideoBitRate` (set recording bit rate),
* Added `shader-toy.recordMaxDuration` (set maximum recording duration),
* Fixed the `shader-toy.recordTargetFramerate` setting,
* Moved the Stats widget to the bottom left, so it doesn't overlap with the GUI.

### CODE REVIEW

File: src/extensions/user_interface/record_video_codec_extension.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    private recordVideoCodec: string;
+
+    public constructor(recordVideoCodec: string) {
+        this.recordVideoCodec = recordVideoCodec;
+    }
+
+    public generateContent(): string {
+        return `"${this.recordVideoCodec}"`;
+    }
+}
```
Comment (Malacath-92) [inline src/extensions/user_interface/record_video_codec_extension.ts:15]: Missing endline at end of file

**Review by Malacath-92 — COMMENTED** [review]
Code-wise I just had the one tiny point, I can fix that after merging as well though. UX wise I'm a bit unsure if the behavior is as intended. When I force a resolution that is bigger than my window size, as a user do I want to
- render at that resolution and resize/zoom the canvas to fit my window
- or have a canvas that is bigger than the window, allowing me to scroll to the side and/or bottom?

Intuitively I would think the former, the behavior I see however is the latter.

File: package.json [hunk 2]
```diff
@@ -113,6 +129,26 @@
...
                 },
+                "shader-toy.recordVideoContainer": {
+                    "type": "string",
+                    "default": "webm",
+                    "description": "Video file container. Currently only webm is supported, but mp4support is coming."
+                },
+                "shader-toy.recordVideoCodec": {
+                    "type": "string",
+                    "default": "vp8",
+                    "description": "Video codec. vp8, vp9, h264 and avc1 are all supported. Default it vp8."
```
Comment (Malacath-92) [inline package.json:140]: Can we use the enum option here? Something along the lines of
```
"shader-toy.recordVideoCodec": {
    "type": "string",
    "default": "vp8",
    "enum": ["vp8", "vp9", "h264", "avc1"],
    "description": "Video codec. vp8, vp9, h264 and avc1 are all supported. Default it vp8."
},
```

**Comment by shadielhajj** [comment]
> Code-wise I just had the one tiny point, I can fix that after merging as well though. UX wise I'm a bit unsure if the behavior is as intended. When I force a resolution that is bigger than my window size, as a user do I want to
> 
> * render at that resolution and resize/zoom the canvas to fit my window
> * or have a canvas that is bigger than the window, allowing me to scroll to the side and/or bottom?
> 
> Intuitively I would think the former, the behavior I see however is the latter.

Yes, fair point. I think this force resolution feature needs more thought. I rolled it back for now, so we can proceed with the rest of the PR.

**Comment by Malacath-92** [comment]
Sorry for the delay, I will merge these two PRs today and test a bit. Hopefully by the evening (8 hours from now) I will be able to finish up and deploy.

**Pull request merged**